library/src/main/java/com/anjlab/android/iab/v3/BillingProcessor.java

```diff
@@ -782,11 +782,11 @@ public boolean consumePurchase(String productId)
 		try
 		{
 			TransactionDetails transaction = getPurchaseTransactionDetails(productId, cachedProducts);
-			if (transaction != null && !TextUtils.isEmpty(transaction.purchaseToken))
+			if (transaction != null && !TextUtils.isEmpty(transaction.purchaseInfo.purchaseData.purchaseToken))
```

Comment:
what is the purpose of this change? can you elaborate

Reply:
The TransactionDetails.purchaseToken field has been deprecated. so used purchaseInfo.purchaseData.purchaseToken field.

Reply:
I bet behaviour will depend on the google play services version installed on a device.
I would do this in a non-breaking manner, checking for both old and new values

Reply:
this function use TransactionDetails instance.
TransactionDetails class constructor.
public TransactionDetails(PurchaseInfo info)
	{
		purchaseInfo = info;
		productId = purchaseInfo.purchaseData.productId;
		orderId = purchaseInfo.purchaseData.orderId;
		purchaseToken = purchaseInfo.purchaseData.purchaseToken;
		purchaseTime = purchaseInfo.purchaseData.purchaseTime;
	}

so used purchaseInfo.purchaseData.purchaseToken field.

Reply:
@serggl Can we Merge ?